feat: add support for _FLOX_SHELL_FORCE (#2620)
flox/flox: cli/flox/doc/flox-activate.md

## Variables used by `flox activate`

`$FLOX_SHELL`, `$FLOX_SHELL_NOCLOBBER`, `$SHELL`
:  When launching an interactive sub-shell, Flox invokes the shell specified in
   `$FLOX_SHELL_NOCLOBBER` if set, then `$FLOX_SHELL` if that is set,
   and finally falls back to invoke `$SHELL` by default.
   Conversely, when performing an "in place" activation
   Flox will attempt to detect its parent shell type
   unless overridden by the `$FLOX_SHELL_NOCLOBBER` variable,
   and if it cannot detect its parent shell type then will

>    produce a script with syntax determined by `$SHELL`.

Reviewer: I don't think this really explains what NOCLOBBER is and for now I'd rather just leave the docs as is

Author: The previous explanation didn't explain what FLOX_SHELL was, it explained how it worked, so I followed suit with my rendition. I agree it would make sense to explain the motivation behind FLOX_SHELL as the means of using flox activate to launch other shells - perhaps adding a sentence to explain that at the beginning would be an improvement?

Author: Updated doc following conversations with @mkenigs ... re-requesting review and resolving conversation.

> `$FLOX_SHELL`, `$FLOX_SHELL_NOCLOBBER`, `$SHELL`
> :  When launching an interactive sub-shell, Flox invokes the shell specified in
>    `$FLOX_SHELL_NOCLOBBER` if set, then `$FLOX_SHELL` if that is set,
>    and finally falls back to invoke `$SHELL` by default.
>    Conversely, when performing an "in place" activation
>    Flox will attempt to detect its parent shell type
>    unless overridden by the `$FLOX_SHELL_NOCLOBBER` variable,
>    and if it cannot detect its parent shell type then will
>    produce a script with syntax determined by `$SHELL`.

Author: ```suggestion
`$FLOX_SHELL`, `$FLOX_SHELL_NOCLOBBER`, `$SHELL`
:  When activating an environment in "interactive" or "in-place" modes,
   Flox will either launch an interactive sub-shell
   or emit commands to configure an already-running (parent) shell.
   In both of these cases Flox needs to know which shell to use,
   and these variables are used to control the selection process.

       * interactive: When launching an interactive sub-shell
         Flox will invoke
         the shell specified in `$FLOX_SHELL_NOCLOBBER` if set,
         then `$FLOX_SHELL` if that is set,
         and finally fall back to invoke `$SHELL` by default.

       * in-place: When performing an "in place" activation
         Flox will attempt to detect its parent shell type
         unless overridden by the `$FLOX_SHELL_NOCLOBBER` variable,
         and if it cannot detect its parent shell type then will
         produce a script with syntax determined by `$SHELL`.
```

## Variables used by `flox activate`

> `$FLOX_SHELL`, `$FLOX_SHELL_NOCLOBBER`, `$SHELL`

Reviewer: If we are going to document FLOX_SHELL_NOCLOBBER, I think the note below is sufficient
```suggestion
`$FLOX_SHELL`, `$SHELL`
```

:  When activating an environment in "interactive" or "in-place" modes,
   Flox will either launch an interactive sub-shell
   or emit commands to configure an already-running (parent) shell.
   In both of these cases Flox needs to know which shell to use,
   and these variables are used to control the selection process.

>        * interactive: When launching an interactive sub-shell

Reviewer: This is the same for command mode, right?

Author: OK will add mention of command mode

>        * interactive: When launching an interactive sub-shell

Author: ```suggestion
       * interactive and command modes: When launching a sub-shell
```

         Flox will invoke
         the shell specified in `$FLOX_SHELL` if set
         or fall back to invoke `$SHELL` by default.

>        * in-place: When performing an "in place" activation

Author: ```suggestion
       * in-place mode: When performing an "in place" activation
```

         Flox will attempt to detect its parent shell type unless overridden by
         the `$FLOX_SHELL` variable,
         and if it cannot detect its parent shell type then will
         produce a script with syntax determined by `$SHELL`.

> 
>     Flox also supports the `$FLOX_SHELL_NOCLOBBER` variable
>     which has the same effect as `$FLOX_SHELL`
>     except that it not cleared from the environment
>     in the subsequent activation.
>     It is used primarily for testing and is not intended as a user-facing feature.

Author: ```suggestion
```

>     Flox also supports the `$FLOX_SHELL_NOCLOBBER` variable
>     which has the same effect as `$FLOX_SHELL`
>     except that it not cleared from the environment
>     in the subsequent activation.
>     It is used primarily for testing and is not intended as a user-facing feature.

Reviewer: I don't think we need to document something and say "but you don't really need to know this"
```suggestion
```

Reviewer 2: +1 for not documenting at all if the only purpose is our testing.

Author: As a rule any variable starting with FLOX ought to be documented. If y'all really don't want this to be a user-facing feature then we should rename this to _FLOX_*.

Reviewer 2: `_FLOX_SHELL_…` makes sense 👍🏻

Author: OK will change to _FLOX_SHELL_FORCE
